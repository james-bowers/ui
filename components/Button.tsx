import * as React from "react";
import styled from "./styled";

type ButtonProps = {
  children: string;
  onClick?: () => Promise<boolean>;
};

let StyledButton = styled.button`
  font-family: ${props => props.theme.fontFamily};
  background: ${props => props.theme.accent};
  color: ${props => props.theme.accentInverted};
  padding: ${props => props.theme.spacing * 3}px;
  font-size: 1em;
  border: 0;
  border-radius: 5px;
  text-transform: uppercase;
  cursor: pointer;
`;

export default (props: ButtonProps) => (
  <StyledButton onClick={props.onClick}>{props.children}</StyledButton>
);
