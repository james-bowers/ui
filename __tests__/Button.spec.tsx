import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

import Button from "../components/Button";

describe("<Button />", () => {
  test("renders Button", () => {
    const tree = shallow(<Button>example button</Button>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  test("renders Button with onClick", () => {
    const tree = shallow(
      <Button
        onClick={() => {
          console.log("clicked");
          return Promise.resolve(true);
        }}
      >
        example button
      </Button>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
