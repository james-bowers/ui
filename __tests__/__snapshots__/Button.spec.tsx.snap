// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders Button 1`] = `
<styled.button>
  example button
</styled.button>
`;

exports[`<Button /> renders Button with onClick 1`] = `
<styled.button
  onClick={[Function]}
>
  example button
</styled.button>
`;
